infrastructure/src/main/kotlin/no/kartverket/matrikkel/bygning/infrastructure/database/repositories/bygning/BygningRepositoryImpl.kt

```diff
@@ -0,0 +1,99 @@
+package no.kartverket.matrikkel.bygning.infrastructure.database.repositories.bygning
+
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
+import com.fasterxml.jackson.module.kotlin.jacksonObjectMapper
+import com.fasterxml.jackson.module.kotlin.readValue
+import no.kartverket.matrikkel.bygning.application.bygning.BygningRepository
+import no.kartverket.matrikkel.bygning.application.models.Bruksenhet
+import no.kartverket.matrikkel.bygning.application.models.ids.BruksenhetBubbleId
+import no.kartverket.matrikkel.bygning.application.models.ids.BruksenhetId
+import no.kartverket.matrikkel.bygning.application.models.ids.BygningId
+import no.kartverket.matrikkel.bygning.infrastructure.database.executeQueryAndMapPreparedStatement
+import no.kartverket.matrikkel.bygning.infrastructure.database.prepareAndExecuteUpdate
+import no.kartverket.matrikkel.bygning.infrastructure.database.withTransaction
+import org.postgresql.util.PGobject
+import java.sql.Timestamp
+import java.util.*
+import javax.sql.DataSource
+
+
+class BygningRepositoryImpl(private val dataSource: DataSource) : BygningRepository {
+    val objectMapper = jacksonObjectMapper()
+        .registerModule(JavaTimeModule())
+        .disable(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS)
+
+    override fun saveBruksenhet(bruksenhet: Bruksenhet) {
+        val bruksenhetToSave = bruksenhet.toDTO()
+
+        return dataSource.withTransaction { connection ->
+            connection.prepareAndExecuteUpdate(
+                """
+                    INSERT INTO bygning.bruksenhet
+                    (id, bruksenhetBubbleId, bygningId, registreringstidspunkt, data)
+                    VALUES (?, ?, ?, ?, ?)
+                """.trimIndent(),
+            ) {
+                it.setObject(
+                    1,
+                    PGobject().apply {
+                        this.type = "uuid"
+                        this.value = bruksenhetToSave.id.toString()
+                    },
+                )
+                it.setLong(2, bruksenhetToSave.bruksenhetBubbleId)
+                it.setObject(
+                    3,
+                    PGobject().apply {
+                        this.type = "uuid"
+                        this.value = bruksenhetToSave.bygningId.toString()
+                    },
+                )
+                it.setTimestamp(4, Timestamp.from(bruksenhetToSave.registreringstidspunkt))
+                it.setObject(
+                    5,
+                    PGobject().apply {
+                        this.type = "jsonb"
+                        this.value = objectMapper.writeValueAsString(bruksenhetToSave)
+                    },
+                )
+            }
+        }
+    }
+
+    override fun getBruksenhetById(id: UUID): Bruksenhet? {
+        return dataSource.executeQueryAndMapPreparedStatement(
```

Comment:
Ryddig hvis params matcher med interfacet den implementerer?
```suggestion
    override fun getBruksenhetById(bruksenhetId: UUID): Bruksenhet? {
        return dataSource.executeQueryAndMapPreparedStatement(
```